tests/test_mapper_inheritance.py

```diff
@@ -171,3 +174,126 @@ def test_entity_inherit_none(self):
+        result = extract_resource_requirements_from_tool(tool)
+        self.assertEqual(result, {})
+
+    def test_tool_with_resource_requirements_mapping(self):
+        cores_req = ResourceRequirement("4", "cores_min")
+        mem_req = ResourceRequirement("8192", "ram_min")
+
+        tool = mock_galaxy.Tool("test_tool_with_resources", resource_requirements=[cores_req, mem_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_uuid_prefix_for_dynamic_tools(self):
+        tool = mock_galaxy.Tool("test_tool", dynamic_tool=mock_galaxy.DynamicTool("12345-67890-abcdef"))
+        tool.tool_type = "interactive"
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+
+    def test_tool_without_uuid_uses_regular_id(self):
+        tool = mock_galaxy.Tool("hisat")
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertEqual(destination.id, "local")
+
+    def test_tool_provided_resource_merging_with_config(self):
+        cores_req = ResourceRequirement("8", "cores_min")
+        mem_req = ResourceRequirement("16384", "ram_min")
+
+        tool = mock_galaxy.Tool("trinity", resource_requirements=[cores_req, mem_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+        self.assertEqual(destination.id, "k8s_environment")
+        # resource requirements override via tool matching
+        assert destination.params["native_spec"] == "--mem 16 --cores 4 --gpus 3"
+
+    def test_default_entity_creation_with_resource_requirements(self):
+        cores_req = ResourceRequirement("2", "cores_min")
+
+        tool = mock_galaxy.Tool("nonexistent_tool", resource_requirements=[cores_req])
+        user = mock_galaxy.User("test_user", "test@test.com")
+        datasets = [mock_galaxy.DatasetAssociation("test", mock_galaxy.Dataset("test.txt", file_size=5 * 1024**3))]
+
+        destination = self._map_to_destination(tool, user, datasets)
+
+        self.assertIsNotNone(destination)
+        assert destination.params["native_spec"] == "--mem 6.0 --cores 2.0 --gpus 1"
```

Comment:
I don't know if the floats are a problem here ?
k8s should support fractional values there, so I'm not sure if we just want to round up to the nearest integer ?

Reply:
Floats should not be a a problem and are supported. Are you running into any issues?

Reply:
No, just didn't know if this is alright with slurm etc

Reply:
Definitely not alright with slurm. In our config we round mem because we expect it to come in as a float. We don’t round cores, which has always been an int, but I guess we should.